src/main/cli/commandLine.ts

```diff
@@ -0,0 +1,120 @@
+import { container } from "readium-desktop/main/di";
+import { extractHostname } from "readium-desktop/utils/hostname";
+import { Store } from "redux";
+import * as yargs from "yargs";
+
+import { OpdsFeedRepository } from "readium-desktop/main/db/repository/opds";
+import { PublicationRepository } from "readium-desktop/main/db/repository/publication";
+import { RootState } from "readium-desktop/main/redux/states";
+import { CatalogService } from "readium-desktop/main/services/catalog";
+
+export interface ICliParam {
+    readonly argv: yargs.Arguments;
+    quit: boolean;
+}
+
+export interface ICli {
+    name: string;
+    fct: (o: ICliParam) => Promise<any>;
+    help: string[];
+}
+
+export const cli: ICli[] = [
+    {
+        name: "_",
+        fct: async ({ argv }) => {
+            let notRead = true;
+
+            argv._.map(async (path) => {
+                // import and read publication
+                const catalogService = container.get("catalog-service") as CatalogService;
+                const publication = await catalogService.importFile(path as string);
+                const store = container.get("store") as Store<RootState>;
+                if (publication && notRead) {
+                    store.dispatch({
+                        type: readerActions.ActionType.OpenRequest,
+                        payload: {
+                            publication: {
+                                identifier: publication.identifier,
+                            },
+                        },
+                    });
+                    notRead = false;
+                }
+                if (path !== "." && !publication && notRead) {
+                    console.error("Publication error, path:", path);
+                }
+            });
+        },
+        help: [],
+    },
+    {
+        name: "import",
+        fct: async ({ argv }) => {
+            const catalogService = container.get("catalog-service") as CatalogService;
+            return await catalogService.importFile(argv.import as string);
+        },
+        help: [
+            "--import PATH",
+            "import epub or lpcl file",
+        ],
+    },
+    {
+        name: "opds",
+        fct: async ({ argv }) => {
+            // extract and save the title and url from opdsFeed
+            // title=http://myurl.com or get TLD and set url
+            const feed = (argv.opds as string).split("=");
+            const url = feed.length === 2 ? feed[1] : feed[0];
+            const title = feed.length === 2 ? feed[0] : extractHostname(url, true);
+            const opdsRepository = container.get("opds-feed-repository") as OpdsFeedRepository;
+            return await opdsRepository.save({ title, url });
+        },
+        help: [
+            "--opds URL",
+            "import opds feed url",
+        ],
+    },
+    {
+        name: "silent",
+        fct: async (param) => {
+            param.quit = true;
```

Comment:
I am concerned about the app lifecycle when using `--silent`, because we have no control over **when** the application exits, when using to `param.quit = true`.
Instead, could we explicitly invoke `process.exit(code)` (with the appropriate exit `code`), **after** the point at which we can guarantee all application jobs have been completed (e.g. importing publications in the database and on the filesystem, opening windows, etc.)

Reply:
I am closing this comment as there are other more relevant and clearer remarks elsewhere.